src/other/broadcasting.jl

```diff
@@ -0,0 +1,94 @@
+struct DummyColumnValue end
+
+Base.broadcastable(df::AbstractDataFrame) = Matrix(df)
+
+Base.maybeview(df::AbstractDataFrame, idxs...) = view(df, idxs...)
+
+function Base.maybeview(df::AbstractDataFrame, idxs)
+    if idxs isa Symbol
+        if !haskey(df, idxs)
+            if !(df isa DataFrame)
+                throw(ArgumentError("column $idxs not found"))
+            end
+            df[idxs] = Vector{DummyColumnValue}(undef, nrow(df))
+            return view(df, [idxs])
+        end
+    end
+    if idxs isa AbstractVector{Symbol}
+        for idx in idxs
+            if !haskey(df, idx)
+                if !(df isa DataFrame)
+                    throw(ArgumentError("column $idxs not found"))
+                end
+                df[idx] = Vector{DummyColumnValue}(undef, nrow(df))
+            end
+        end
+    end
+    view(df, idxs)
+end
+
+function Base.copyto!(df::SubDataFrame, bc)
+    m = Base.materialize(bc)
+    if ncol(df) == 1 && eltype(df[1]) === DummyColumnValue
+        if maximum(ndims.(Base.Broadcast.flatten(bc).args)) > 1
+            throw(DimensionMismatch("cannot broadcast array to have fewer dimensions"))
+        end
+    end
+    for col in axes(df, 2)
+        if eltype(df[col]) === DummyColumnValue
+            colname = names(df)[col]
+            T = reduce(promote_type, typeof.(view(m, :, col)))
+            parent(df)[colname] = Tables.allocatecolumn(T, nrow(df))
+        end
+        df[col] .= view(m, :, col)
```

Comment:
This looks wrong to me.  Are you sure the dimensions work out to broadcast between `m` and `df` if you do it columnwise like this?  It feels like you're missing an error condition or two, but I haven't completely worked it out.

Reply:
Oh, and remember that this is the general `copyto!` method — it's what will get called for arbitrary iterables and you're not guaranteed to have something `view`able.

Reply:
This is a very good point. I think that the signature of `copyto!` should be restricted. We accept here only such `bc` that are two-dimensional. Of course maybe this should be changed in general if we can avoid `m = Base.materialize(bc)` (given your comment above). I used view only for performance. It can be dropped.